### DISCUSSION TITLE
Quick Settings Menu Design

### DISCUSSION THREAD

**Discussion by danirabbit** [elementary/wingpanel#446] [Ideas]
# Problem

If we want to start thinking about responsive/mobile, the current panel design has a couple of issues:

Each group of settings necessitates an icon to reach it, which means the more groups of settings we want, the more icons we need. This is already awkward on desktop for example, where we have a placeholder power icon so you can see the battery status of peripherals. On mobile we have limited horizontal space, so if we want to indicate things like portal usage etc, we need to avoid icons that aren't indicating important information.

Since each menu is triggered by a separate icon, you need much more precise pointing to activate the correct menu. On desktop we compensate for that with scrubbing, and we could do something similar on mobile (I think Ubuntu Touch does this), but it is a much less granular gesture to swipe or tap a large activatable area to reach a single menu.

There is already a desire for some controls and indicators to be transient, for example night light and privacy/portal indicators. At the moment, night light is not accessible from the panel while not activated. It's also unclear how to categorize settings such as dark mode. Some privacy/portal indicators make sense to be grouped with other settings like microphone, but it's unclear how to group others like webcam or screen recording or location. There is also might be utility in quick controls that don't necessarily indicate anything like screen casting or controlling IoT devices

# Proposal

It might make more sense to move to a quick settings menu style design with a singular menu for accessing these types of controls. This would eliminate the need for each group of controls to require an associated icon and give us a single click/tap/drag target to open the menu. It's worth considering that this could also help users on desktop with different kinds of pointing devices or motor control ability.

We could be a lot stingier with only showing icons that indicate a change in state from what is expected or that need attention. For example, iOS doesn't show a volume indicator at all times. We might not need to show an indicator for things that are in their expected state, like bluetooth enable/disable.

We could show indicators for things that don't necessarily have quick controls associated or are hard to group like privacy/portals

We could include quick controls for things that don't need an indicator like nightlight and dark mode, peripheral battery status, and other potential controls like IoT devices, screen casting, etc.

# Prior Art

### Windows
![iu4](https://user-images.githubusercontent.com/7277719/162799775-7788fa40-d1a3-4ee2-8b2e-96e5cc5a96bf.jpeg)

### macOS
![iu2](https://user-images.githubusercontent.com/7277719/162799780-b42d6ed1-6fff-4042-9755-5a0f45655606.jpeg)

### GNOME
![iu](https://user-images.githubusercontent.com/7277719/162799819-669eb614-0340-469b-ab20-566897b944ac.jpeg)

### Chrome OS
![AB0E1660-3C14-442E-8494-C3B3B5DBAD0A](https://user-images.githubusercontent.com/7277719/222035565-b9740214-6a8c-4cff-a0ed-25d1244ea38e.jpeg)

**Comment by nothingneko** [comment]
Have you seen GNOME's new design Danielle?
https://gitlab.gnome.org/Teams/Design/os-mockups/-/commit/f07e260f25afb96dcfa7fcf4401a47c73bdc8f72
![image](https://user-images.githubusercontent.com/37665485/162803229-a10589ab-4c49-40a1-9e97-f756d73579aa.png)

**Reply by danirabbit** [reply]
Thanks for sharing this, I had not! This is much more comprehensive

**Reply by nothingneko** [reply]
No problem! My team is honestly just waiting for these to drop in GNOME 43, I'd love to see it in Pantheon too!

**Reply by ghost** [reply]
It really is a very beautiful design, and seeing the settings like on/off wifi bluetooth all grouped together in a single click in the top corner of the elementary os bar will give a very attractive and more organized design, just like with Windows and Mac OS , I support the idea.

**Reply by Lucy1958** [reply]
Why reinvent the wheel?  Use Gnome's new configuration  settings.

**Reply by danirabbit** [reply]
> Why reinvent the wheel? Use Gnome's new configuration settings.

elementary OS doesn’t use GNOME so that’s not possible. Pantheon is it’s own DE like XFCE or KDE

**Reply by homerograco** [reply]
These settings in elementaryOS style (less rounded corners, horizontal dividers, sharp icons and font) would be great.

**Reply by radical4ever** [reply]
I think they added a new mockup of how they will look with the app inidcators (a.k.a system tray icons) that would be awesome. they made a separate space from the rest of the settings and not directly to the top bar wich i personally think is less messy than having them all dispersed in the top bar.

**Comment by ghost** [comment]


**Reply by danirabbit** [reply]
Please try to keep things on topic. You can open new discussions about onboarding here: https://github.com/elementary/onboarding/discussions

**Comment by popvladaurel** [comment]
Cool cool, as long as it does not change the scrolling-over-indicators behavior: changing volume / brightness by scrolling over the volume / battery indicators, or middle-click volume indicator to mute.

**Reply by nothingneko** [reply]
The scrolling is honestly amazing UX

**Reply by danirabbit** [reply]
Hm yeah that's an interesting counter-point for hiding some indicators. Perhaps we could either always persist icons who have scroll controls or make it configurable

**Reply by nothingneko** [reply]
I think making it configurable is the best choice

**Reply by homerograco** [reply]
That and a way of dragging them into or out of the "indicators group" would be lovely

**Reply by flodavid** [reply]
Maybe the volume icon should be shown in the panel only when some audio is playing

**Comment by preetYOW** [comment]
MacOs provides an option to enable the icon triggers for most items despite their presence in control panel. 
Eg. you could change wifi settings from control panel or (can enable wifi icon fro settings) which will allow you to change wifi settings directly from menu bar.

Pantheon could implement something similar. eg. One setting to allow menu bar icons to be grouped (clicking any menubar icon opening the panel)/ or present implimentation (clicking specific menubar icon will open ony that menu).

**Reply by danirabbit** [reply]
I'm hesitant about this because of the effort required to maintain two different implementations. I think we could leverage the plugin system of the panel to develop a quick menu and have it be installable as an option until it's mature, but I don't think the team would like to maintain both the quick menu and another set of indicators long term

**Comment by Lucy1958** [comment]
Couldn't a code modification of the new Gnome settings be accomplished for Pantheon?  Seems like the basic concept would be worth pursuing.12:38 AM, April 12, 2022, "Danielle Foré" ***@***.***>:

Why reinvent the wheel? Use Gnome's new configuration settings.

elementary OS doesn’t use GNOME so that’s not possible. Pantheon is it’s own DE like XFCE or KDE

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
-- Sent from Yandex Mail for mobile

**Reply by danirabbit** [reply]
No, it's not possible. GNOME Shell is completely dissimilar from Pantheon. It uses it's own internal toolkit whereas Pantheon uses Gtk

**Comment by Lucy1958** [comment]
Was talking concept not specific code programming and/or execution. 4:48 PM, April 12, 2022, "Danielle Foré" ***@***.***>:
No, it's not possible. GNOME Shell is completely dissimilar from Pantheon. It uses it's own internal toolkit whereas Pantheon uses Gtk

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
-- Sent from Yandex Mail for mobile

**Comment by nothingneko** [comment]
Is it planned to include power profiles in the quick settings? Iirc power profiles were planned for OS7

**Reply by danirabbit** [reply]
Power profiles shipped in OS 7. There’s some work on improving them happening right now that might resolve an issue we had with showing them in the panel

**Comment by JohnDumpling** [comment]
I have made a mockup for a possible redesign of the control center and notifications menu. (Please ignore the wrong spacing between some elements and the placeholder icons 🙂). I decided to group all of the separate submenus into one, which can be opened by clicking on any of the right-hand-side icons in the Wingpanel (those could be grouped similarly to how Gnome does it). The menu is split into two tabs: Controls - for access to quick settings and Notifications - for notifications. You can switch between these two views by clicking on the buttons in the menu's header bar.

**Controls menu**
Contains options for switching user accounts, power off / restart / suspend / log-out, then toggles for wi-fi, bluetooth, power profiles, volume and brightness sliders etc. Active toggles are indicated by a green dot, inactive by a grey one. More detailed options can be accessed by clicking on the arrow-down button on the right side of some toggles (sound output selection, night light adjustment etc.). There is even a Power "widget" that shows the battery status of connected devices (these "widgets" could be potentially used for cloud sync or other interesting functionalities). All the toggles can be rearranged, hidden or made visible by the user.
![Controls menu](https://user-images.githubusercontent.com/89086803/175647833-d5d00f65-4cbb-4f67-8e5a-9904eb62bfb9.png)

**Notifications menu**
Displays notifications in a similar way to the current implementation. I decided to separate this view from the Controls tab to make the menu look more compact on smaller screens.
![Notifications menu](https://user-images.githubusercontent.com/89086803/175647827-2ed4685c-25b6-4d64-b3ef-aa7684351133.png)

Any feedback is appreciated!

**Reply by radical4ever** [reply]
Well, this is what im talking about of something unique for elementary os!

**Reply by janxkoci** [reply]
FYI the same idea is used in the Raven panel of the Budgie desktop.

**Comment by Lucy1958** [comment]
Your implementation looks great!  Could you add a control for printer setup as well as a notification (jobs completed, ink levels, etc.) for printers?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Romchec** [comment]
I think that the old implementation of panel is better than single menu cluttered with buttons. Even on 1280x720 there still a lot of free space, because there is not tray icons by default. Also how many users will use elementary on mobile-sized screen? Why hundreds of PC users should  make compromises because of a minority of 5 people from all over the world.  Recently, due to the widespread optimization of the PC interface for touch control, using a PC is becoming less convenient.

**Comment by DSLitvinov** [comment]
I'm probably not the only one who doesn't see the point in spending man-hours adapting the interface for mobile devices? Just ask yourself what percentage of the mobile device market does Gnome represent? And he has been developing the idea of a hybrid environment for more than 10 years. As a user of Win11, Linux (plasma 5, Elementary OS) and Mac, I can say that this curtain on a desktop has no advantages over separate controls. Instead of one action, you force the user to do two or three. If the curtain is not customizable, then it will be a failure.

**Reply by flodavid** [reply]
How do you expect eOS (or even Linux as a whole) to have mobile users if it is never adapted to this usage ? The work has to be done as some point if we want it to have a chance of developing some kind of user base there

**Reply by janxkoci** [reply]
GNOME and KDE work on mobile today. What's the market share? Nil..

**Comment by jlnr** [comment]
I came here from the latest blog post:

> In recent years every mainstream operating system has adopted a Quick Settings menu design which lends itself much better to responsive contexts as well.

In my opinion, neither "everyone else is doing it" nor "it's necessary on mobile" should have any effect on the desktop UI. There is no mobile elementary OS right now; I wish it existed, but wouldn't wingpanel + dock have to be replaced with a distinct mobile UI anyway?

That makes the situation different from macOS, which often emulates iOS to the detriment of desktop usability (see new System Settings app 🙄). Windows is yet another story, with a system tray that is overflowing with third-party noise, which their new design avoids. But elementary has already dropped support for arbitrary indicators.

Both Windows and macOS also have a system tray that competes for space with other elements on the screen: app menus on macOS, task bar on Windows. That's not the case in elementary OS either.

**Reply by jlnr** [reply]
Re transient items:
Night Light could be combined with a dark/light mode toggle as a new "Display" icon, maybe with a slider for screen backlight (where possible) and a toggle for mirroring or extending the external screen (if any). I'd personally prefer that to icons that appear and disappear over the course of the day.

**Reply by teamcons** [reply]
Hard agree.
From a user interface standpoint, separate items is awesome : Me want sound, me click sound icon, me has only few sounds items.
A quick settings menu, people have to get that by clicking a whole spam of icons they suddenly get a whole spam of options for the one thing they want. Its nice, but it is more noisy

Mobile is a nonfactor. How much of the userbase needs this for a mobile ?
What other OS does is a nonfactor too. If its that important to do the others, better scrape the whole DE and just use GNOME - Pantheon deserves to have its own nice things too

Besides, resources are limited, why do things for the heck of it ?
Im not necessarily against quick settings at all costs but i dont really get why transition to that design ?

**Comment by stsdc** [comment]
Just something I found on the net. I'll just leave it here https://x.com/him_uiux/status/2007021579784614356
<img width="900" height="642" alt="obraz" src="https://github.com/user-attachments/assets/9ffc2a63-3e4d-47d6-b74a-2326cc7efa1f" />

**Comment by kurisubrooks** [comment]
Continuing on from some previous mockups I've done in other areas of the OS, I thought I'd give this one some thought as well. 

Taking into consideration a possible extension into the mobile space, as well as some more modern direction while also keeping the elementary OS "feel" of brightly coloured icons. 

I also wanted to provide some options for @danirabbit as she mentions that she'd like to see certain panel controls become transient/context-aware. I've included various methods of separation in order to retain an always-visible panel for important indicators like the time/date, notifications, and quick settings. I've also included an area for core system controls, like Wi-Fi, sound, and power. Though if Dani wishes, this could be hidden by default on mobile.

I also separated out an area for less important/third-party indicators, like Discord or system stats into its own area. My thought was that this could be collapsible if there were too many items, raising themselves to be shown if they're in an "active" state, though I'm not sure APIs exist for this.

<img height="473"  alt="image" src="https://github.com/user-attachments/assets/06e791b6-97ef-4c7e-84be-3829e860ace2" />

I also put some thought into the quick settings and pane views, which I wanted to feel more airy, and is inline with the newer design paradigms/direction of elementary. Adding some more colour where there currently is none by using the 32px icons, as well as making use of the user's accent colour throughout, I feel this feels more fun and interactive than the current greyscale/flat panel design. "So good you want to lick it."

<img width="1080" height="696" alt="Screenshot 2026-05-07 at 3 22 02 pm" src="https://github.com/user-attachments/assets/cefc01c6-3fd4-47d9-889a-7c0c83911e90" />

**Reply by zeebok** [reply]
I really love these designs! Clean and colorful, with the accent color visible! I do wonder if it is worth maintaining a hypothetical submenu when clicking the button instead of opening the appropriate indicator menu or the system settings themselves?

**Reply by kurisubrooks** [reply]
I designed the sub-menu with the intention that these panes would reuse the applet/window across entry points, so we won't have to maintain separate implementations. Necessary to facilitate an OS where the top indicators may not always be visible/accessible (like mobile, or if Danielle goes ahead with the transient indicators system), I felt this was the best solution to reduce maintainer fatigue.

<img width="1369" height="444" alt="Screenshot 2026-05-08 at 11 20 19 am" src="https://github.com/user-attachments/assets/da0fe550-bd53-48c7-a502-7490d79e953b" />

**Reply by kurisubrooks** [reply]
For reference, it looks like Apple does the same in Tahoe:

(Menu bar applet)
<img width="374" height="325" alt="image" src="https://github.com/user-attachments/assets/32acb850-d805-428a-8a07-e42d38d72426" />

(Control Centre applet)
<img width="654" height="364" alt="Frame 88" src="https://github.com/user-attachments/assets/5d34623d-82f0-4bc2-ab02-a6e3161690b7" />

**Reply by stsdc** [reply]
I love the fact that buttons are not flat, they look like bubbles from bubble wrap :sweat_smile:
